### ISSUE TITLE
Program runs then dies

### ISSUE THREAD

**Issue by ardee00** [Klocman/Bulk-Crap-Uninstaller#294]
On a Win7 Ultimate system with 64GB RAM and vast free disk space, the program runs for a few seconds then disappears.  The system has a very large number of programs installed (perhaps 1000); can BCUninstaller handle such a large load?
If you want, I will be a beta tester as I am anxious to use this fantastic-looking software.

**Comment by Klocman** [comment]
Try the test build I posted in #291

**Comment by ardee00** [comment]
Thank you for the response! Unfortunately, the installer for that version won’t even start on my PC. (I do have all the Microsoft redeistributables installed.)

**Comment by ardee00** [comment]
I just realized that the .zip file in #291 downloads to my PC as a tiny file (under 1 MB).  I imagine it's supposed to be a lot bigger -- is there another copy to try?

**Comment by Klocman** [comment]
That's just the launcher .exe, use it to replace your existing bcuninstaller.exe in the root directory (not in the subdirectories with many files). If that doesn't help, try running bcuninstaller.exe directly from the win-x64 folder.

**Comment by ardee00** [comment]
Yeah, that dawned on me when I was away from my computer.  I should have deduced that right away!

Anyway, replacing the .exe from the installer with the one from the .zip results in the same symptom: it runs nicely for a second or two then just quits.

When you say to run it from the win-x64 folder, do you mean the .exe that put there by the installer, or the replacement one?

Thank you very much!

**Comment by ardee00** [comment]
So I tried both ways; using the original exe in the win-x64, the program runs for several seconds longer than before, but ends up quitting before it's done scanning.  The replacement exe won't run at all from that folder (but you probably already knew that).

My offer to be a beta tester still stands.

**Comment by Klocman** [comment]
After it stops working, check the x64 directory for any error report .zip files, send the latest one if you see any. If not, go to system event viewer > application log, and look for errors related to BCUninstaller.exe. Send whatever you can find.

**Comment by ardee00** [comment]
There is nothing in the event logs, and no zip file.  There _is_ a .log file in the x64 directory, but all it has is multiple instances of

--------------------------------------------------
01:30:17 p - Application startup

with different times.

I think my system is a good "extreme conditions" testbed, because of the huge numbers of applications installed and the multiple hard drive letters mounted.  I'll keep trying if you will.  THANK YOU!

**Comment by Klocman** [comment]
Try using these settings. If it works, try turning on different detection options and see what causes it to fail.
[BCUninstaller.zip](https://github.com/Klocman/Bulk-Crap-Uninstaller/files/6566810/BCUninstaller.zip)

**Comment by ardee00** [comment]
That worked!  I placed the settings file into the same directory as the exe, and things ran as expected.  I will experiment with different options and get back to you.  Thanks so much.

**Comment by ardee00** [comment]
OK, here's the update: it was definitely the Registry scanning that caused the crash.  I have plenty of other evidence that the registry on this PC has issues.  I am hesitant to use any "registry repair"-type utilities, but if you suggest one you're confident in, I'll try it.

May I add, wow, what a program you've created!  The level of attention to detail and the level of user feedback/control are extraordinary.  I have hundreds of applications installed on my D: drive (to preserve space on my SSD), and this your program located them despite my having disabled registry scanning.

I encountered one crash, which occurred after I hit the down-arrow after changing the sort order of the list of applications.  (I did that because the one I'd had selected was no longer visible and I thought that, like some applications, it was still selected but scrolled out of view and hitting the up or down arrow key would scroll it back into view.)  Your program provided details about the crash and offered to send a report to you -- which I accepted.

As a long-time programmer, I recognize the rare mastery of Windows internals you have.  Would it be possible to contact you offline to see if I can hire you to help resolve some (probably complex) issues that prevent Windows Update and System Restore from running as well as prevent me from upgrading to Windows 10 as an upgrade (or even reinstalling Windows 7)?

Thank you very much,

  -Roy E.

**Comment by Klocman** [comment]
Thank you for the follow-up!

If BCU completely dies during registry scan with no errors then there's something very wrong with the registry. You should get at least a crash report box if something goes wrong. If it hard crashes then it's most likely an unrecoverable error originating from outside BCU, indicating some sort of system issue.

You can try [this utility](https://www.bleepingcomputer.com/download/windows-repair-all-in-one/), if that doesn't help then try running chkdsk, sfc /scannow, and repair with dism. If none of these help with your system then it's best to just backup your user folder and any important files, and do a clean reinstall of your OS.

**Comment by ardee00** [comment]
Thank you Klocman!  chkdsk is fine and sfc /scannow says there are issues but it can't identify or fix them. (I don't recall what happened last time I tried dism, except that it didn't help, but I may try again.)  The repair program you mentioned is interesting: it found numerous missing and broken reparse points; it offers to repair them but I'm wary because it doesn't say HOW it would do that, nor whether doing that could be unsafe (or unhelpful at best).

I can never do a clean (re)install because I have about 1,000 installed applications, many on D:, many going back years, and many that were only free on the day originally offered.  Also, my PC's USB ports, and NVMe slot aren't recognized by the Windows 7 installer, but installing Windows 10 would mean losing many things, including programs I wrote years ago that won't run under 10.
